syntax = "proto3";

package second.protos;
import "second/protos/preprocess.proto";
import "second/protos/sampler.proto";

message InputReader {
  uint32 batch_size = 1;
  message Dataset {
    string kitti_info_path = 1;
    string kitti_root_path = 2;
    string dataset_class_name = 3; // support KittiDataset and NuScenesDataset
  }
  Dataset dataset = 2;
  message Preprocess {
    bool shuffle_points = 1;
    uint32 max_number_of_voxels = 2;
    repeated float groundtruth_localization_noise_std = 3;
    repeated float groundtruth_rotation_uniform_noise = 4;
    repeated float global_rotation_uniform_noise = 5;
    repeated float global_scaling_uniform_noise = 6;
    repeated float global_translate_noise_std = 7;
    bool remove_unknown_examples = 8;
    uint32 num_workers = 9;
    float anchor_area_threshold = 10;
    bool remove_points_after_sample = 11;
    float groundtruth_points_drop_percentage = 12;
    uint32 groundtruth_drop_max_keep_points = 13;
    bool remove_environment = 14;
    repeated float global_random_rotation_range_per_object = 15;
    repeated DatabasePreprocessingStep database_prep_steps = 16;
    Sampler database_sampler = 17;
    int64 min_num_of_points_in_gt = 19; // gt boxes contains less than this will be ignored.
    bool random_flip_x = 20;
    bool random_flip_y = 21;
    float sample_importance = 22;
  }
  Preprocess preprocess = 3;
  uint32 max_num_epochs = 4; // deprecated
  uint32 prefetch_size = 5;  // deprecated
  message CumLCWrapper {
    string lc_policy = 1; 
    uint32 lc_horizon = 2;
    uint32 init_lidar_num_beams = 3;
    message SparsifyReturn {
      float max_height = 1;
      uint32 pos_intensity_thresh = 2;
      float neg_subsampling_rate = 3;
    }
    SparsifyReturn sparsify_return = 4;
  }
  CumLCWrapper cum_lc_wrapper = 6;
}
